src/bsoncxx/types/value.cpp

```diff
@@ -96,11 +102,19 @@ bsoncxx::type value::type() const {
 #undef BSONCXX_ENUM
 
 bool operator==(const value& lhs, const value& rhs) {
-    if (lhs.type() != rhs.type()) {
+    if (!lhs.type() && !rhs.type()) {
+        return true;
```

Comment:
I just assumed two invalid types would be equal. Maybe there is some benefit to them not being equal, though I'm not sure what it would be.